pkg/controller/replication/replication_controller.go

```diff
@@ -266,19 +272,42 @@ func (rm *ReplicationManager) getPodController(pod *api.Pod) *api.ReplicationCon
 		// the first.
 		glog.Errorf("user error! more than one replication controller is selecting pods with labels: %+v", pod.Labels)
 		sort.Sort(OverlappingControllers(controllers))
+
 	}
 
-	// update lookup cache
-	rm.lookupCache.Update(pod, controllers[0])
+	ret_rc = controllers[0]
+
+	// check ownerReference.Controller and UID to determine if the rc controls the pod
+	// in case of multiple controllers match the pod's label
+	if controllerRef != nil {
+		ret_rc = nil
+		for _, rc := range controllers {
+			if controllerRef.UID == rc.ObjectMeta.UID {
+				ret_rc = rc
+				break
+			}
+		}
+	}
+
+	if ret_rc == controllers[0] {
```

Comment:
consider defering this last two checks so you can just return from the above loop.